### DISCUSSION TITLE
Link Custom Representation

### DISCUSSION THREAD

**Discussion by syedhabib53** [BeanieODM/beanie#359] [Question]
Hi Roman,

Right now `Link` (`to_dict` method) representation is looks like 

```json
{
    "id": "<id of the link>",
    "collection": "<collection name of the link>"
}
```

I am trying to customize the representation which means `to_dict` method to achieve only getting `id` of the link, using custom link class which looks below. 

```python
from typing import TypeVar, Generic

from beanie import Link

RT = TypeVar("RT")


class ReprLink(Link, Generic[RT]):
    def to_dict(self):
        return str(self.ref.id)

```

My question is, Is there any other(easy 🙄) way to get ride of this, without having custom class?!

**Comment by roman-right** [comment]
Hey! You can try projections, which will help you construct the correct data format on the fetch step. But I'd say your approach looks better, as with projections you will not be able to use links as links in the fetched documents.

**Reply by syedhabib53** [reply]
Yeah. I agree with you. For that I need to define my model with my custom class. I would prefer, don't want to add more custom complexity with `Beanie`. I would like as it is with my code. 😊
